compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -69,8 +69,8 @@ object DesugarEnums {
 
   /** Add implied flags to an enum class or an enum case */
   def addEnumFlags(cdef: TypeDef)(implicit ctx: Context): TypeDef =
-    if (cdef.mods.isEnumClass) cdef.withMods(cdef.mods.withFlags(cdef.mods.flags | Abstract | Sealed))
-    else if (isEnumCase(cdef)) cdef.withMods(cdef.mods.withFlags(cdef.mods.flags | Final))
+    if (cdef.mods.isEnumClass) cdef.withMods(cdef.mods.toTypeFlags | Abstract | Sealed)
```

Comment:
Note the `toTypeFlags` here. It is to prevent a `illegal flagset combination` assertion failure that occurs when combining `TermFlags` with `TypeFlags`. For example when parsing:
```scala
lazy enum Foo {}
```